### DISCUSSION TITLE
如何在target编译的时候，连接xrepo在其他目录下生成的lib

### DISCUSSION THREAD

**Discussion by scially** [xmake-io/xmake#2998] [Q&A]
以xmake-repo中的openscenegraph库为例，这个库在安装后，还会有一个osgPlugins目录，这里面的lib是不会自动连接的，我想对在target编译的时候连接其中的osgPlugins下面的某几个lib，改怎么弄？
![image](https://user-images.githubusercontent.com/22127608/198820868-02f4e9f7-0679-4054-a511-20dfe94aa8b0.png)

**Comment by scially** [comment]
在其他代码里找到了这个方法，可是在官网文档上没找到 target:pkg api说明，文档是不是漏了……
```
    before_link(function (target)
        local osg_pkg = target:pkg("openscenegraph")
        local osg_basepath = osg_pkg:installdir() .. "/lib/osgPlugins-" .. osg_pkg:version()
        osg_pkg:add("linkdirs", osg_basepath)
        osg_pkg:add("libfiles", osg_basepath .. "osgdb_osg.lib")
        osg_pkg:add("links", "osgdb_osg")
    end)
```

**Comment by waruqi** [comment]
add_links add_linkdirs 在 package 里，可以看下其他包
